scripts-components/in-view/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, ComponentInitializationError } from '@mangoweb/scripts-base'
+import { IsInView, isInView, isOutOfView, isSupported } from 'isinview'
+
+interface InViewData {
+	targets?: string
+	threshold?: number
+
+	isAboveViewClass?: string
+	isInViewClass?: string
+	isBelowViewClass?: string
+	isSeenClass?: string
+}
+
+export class InView extends Component<InViewData> {
+	static componentName = 'InView'
+
+	protected readonly targets: IsInView.Target
+	protected readonly options: Partial<IsInView.Options> = {}
+
+	protected readonly isAboveViewClass: string
+	protected readonly isInViewClass: string
+	protected readonly isBelowViewClass: string
+	protected readonly isSeenClass: string
+
+	public constructor(el: HTMLElement, data: InViewData) {
+		super(el, data)
+
+		if (!isSupported()) {
+			throw new ComponentInitializationError('InView component not supported. Add IntersectionObserver polyfill.')
```

Comment:
👍

Reply:
Teď jsem si uvědomil, že nevím, jak moc velký problém způsobuje vyhození této výjimky. Projde inicializace ostatních komponent v pohodě? Tohle totiž nemusíš chytit během vývoje. (dotaz spíš na @JirkaVebr)

Reply:
Ta výjimka se vyhodí v IE a ve starém Safari, pokud kódér nenasadil polyfill. Způsobí to akorát to, že na webu nepojede `InView`, takže by se neměl úplně rozpadnout.

Testovat weby ve všech prohlížečích tak jako tak musíme, takže mi přijde ok, že tady vyhazuji jen tu výjimku a nic víc.

Nebo co víc myslíš, že by jsme mohli udělat? (Nasazovat polyfill přímo do komponenty se mi tentokrát moc nechtělo, protože většina browserů podporu má, takže by to vedlo ke zbytečnému objemu. Navíc nasadit polyfill třetí strany by mělo být easy.)

Reply:
Podle mě si nerozumíte. @enzy se ptá, jestli tu výjimku něco spolehlivě chytí, na což je odpověď že ano, pokud ji vyhodíš v `constructor` nebo v `init`, chytí ji `componentInitializer`. Ostatních komponent se to tedy nedotkne.